packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,373 @@
+| children            | `string`   | No       | undefined | The text to be rendered as CheckboxLabel                                                                                                                 |
+| helperText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| isChecked           | `boolean`  | No       | false     | state of the checkbox, if explicitly set the checkbox will be controlled                                                                                 |
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will be uncontrolled                                                            |
+| onChange            | `Function` | No       | undefined | The function to be called when the checkbox state changes.                                                                                               |
+| isIndeterminate     | `boolean`  | No       | false     | mixed state of the checkbox                                                                                                                              |
+| name (html native)  | `string`   | No       | undefined | The name of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)            |
+| value (html native) | `string`   | No       | undefined | The value of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |
+| isDisabled          | `boolean`  | No       | `false`   | Control whether the checkbox is disabled or not.                                                                                                         |
+| isReadOnly          | `boolean`  | No       | `false`   | Control whether the checkbox is readonly or not.                                                                                                         |
+| isInvalid           | `boolean`  | No       | `false`   | Control whether the checkbox is invalid or not.                                                                                                          |
+| isRequired          | `boolean`  | No       | `false`   | Control whether the checkbox is required or not.                                                                                                         |
+
+### Examples:
+
+#### Basic
+
+```tsx
+// Label text
+<Checkbox>is blade awesome?</Checkbox>
+// Helper text
+<Checkbox helperText="Hint: yes">is blade awesome?</Checkbox>
+// uncontrolled
+<Checkbox defaultChecked={true}>is blade awesome?</Checkbox>
+// controlled
+<Checkbox isChecked={true | false} onChange={isChecked => {}}>is blade awesome?</Checkbox>
+```
+
+#### Validations
```

Comment:
Some of these validations seem a bit complex to understand at first glance. You've explained it well here but our API won't be able to communicate it when a consumer is using it.  Lets discuss if we can reduce some complexity here?

Reply:
We will need these anyway. and now we decided to have validationState prop so it will stay as is. Let me know if you have any ideas to reduce the complexity further.